Add QNAP QSW sensors for each port

Noltari opened the pull request:
## Proposed change
Add QNAP QSW binary sensors for each port.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/25621

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

Noltari committed:
qnap_qsw: add individual port sensors

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari commented:
Rebased on top of https://github.com/home-assistant/core/pull/76522 https://github.com/home-assistant/core/commit/873b903cf0217c48049e32c45f7d5b115078b831
Added changes suggested by @bdraco and @balloob for binary sensors.

bdraco reviewed (commented):
Please resolve conflicts

bdraco committed:
Merge branch 'dev' into qnap-qsw-port-sensors

bdraco commented on homeassistant/components/qnap_qsw/sensor.py:
```diff
@@ -152,20 +157,187 @@ class QswSensorEntityDescription(SensorEntityDescription, QswEntityDescription):
+            entities.append(QswSensor(coordinator, description, entry))
+
+    for description in LACP_PORT_SENSOR_TYPES:
+        if (
+            description.key in coordinator.data
+            and QSD_LACP_PORTS in coordinator.data[description.key]
+        ):
```
I'd reverse this conditional and continue here and below to reduce the indentation level.

Noltari replied:
Done in ee03a0308519d6f7c7b3765b99288dcf642fa22d

bdraco commented on homeassistant/components/qnap_qsw/sensor.py:
```diff
@@ -178,21 +350,23 @@ def __init__(
-        self._attr_unique_id = (
-            f"{entry.unique_id}_{description.key}_{description.subkey}"
-        )
+        self._attr_unique_id = f"{entry.unique_id}_{description.key}{description.sep_key}{description.subkey}"
```
Is this changing any existing sensors?

Noltari replied:
> Is this changing any existing sensors?

Nope, it isn't because sep_key is `_` by default, so it remains the same for the existing sensors.

Noltari committed:
qnap_qsw: sensor: switch DATA_BYTES to UnitOfInformation

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari committed:
qnap_qsw: sensor: improve readability

Introduces changes suggested by @bdraco to improve readability and reduce
indentation.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

bdraco commented:
Do any of the docs need to be updated for this?

Noltari commented:
> Are there any docs that need to be updated for this?

I guess I can add the new port sensors to the existing sensors section...
https://www.home-assistant.io/integrations/qnap_qsw#sensors

bdraco commented:
Should be good to merge once docs PR is open

Noltari commented:
> Should be good to merge once docs PR is open

Docs PR created: https://github.com/home-assistant/home-assistant.io/pull/25621

bdraco committed:
Merge branch 'dev' into qnap-qsw-port-sensors
